File: app/src/main/java/com/u1fukui/bbs/utils/ToastUtils.java
package com.u1fukui.bbs.utils;


import android.content.Context;
import android.support.annotation.StringRes;
import android.widget.Toast;

public class ToastUtils {

    private final Context context;

    private Toast toast;

    public ToastUtils(Context context) {
        this.context = context;
    }

    public void showToast(@StringRes int messageId) {
        showToast(context.getString(messageId));
    }

    public void showToast(CharSequence message) {
        if (toast != null) {
            toast.cancel();
        }
        toast = Toast.makeText(context, message, Toast.LENGTH_SHORT);
        toast.show();
    }
}

File: app/src/main/java/com/u1fukui/bbs/utils/ToastUtils.kt
package com.u1fukui.bbs.utils


import android.content.Context
import android.support.annotation.StringRes
import android.widget.Toast
import com.u1fukui.bbs.network.ApiErrorMesageResolver

class ToastUtils(private val context: Context) {

    private var toast: Toast? = null

    fun showToast(@StringRes messageId: Int) {
        showToast(context.getString(messageId))
    }

    fun showApiError(throwable: Throwable) {
        showToast(ApiErrorMesageResolver.getMessage(throwable))
    }

    fun showToast(message: CharSequence) {
        if (toast != null) {
            toast!!.cancel()
        }
        toast = Toast.makeText(context, message, Toast.LENGTH_SHORT)
        toast!!.show()
    }
}
